src/auth/integration-tests/src/lib.rs

```diff
@@ -115,3 +120,118 @@ pub async fn api_key() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+pub async fn workload_identity_provider_url_sourced() -> anyhow::Result<()> {
+    let project = std::env::var("GOOGLE_CLOUD_PROJECT").expect("GOOGLE_CLOUD_PROJECT not set");
+    let audience = get_oidc_audience();
+    let service_account = get_byoid_service_account();
+    let service_account_data =
+        serde_json::from_value::<HashMap<String, String>>(service_account.clone())
+            .expect("failed to read service account data as map");
+    let client_email = service_account_data
+        .get("client_email")
+        .expect("missing client_email");
+
+    let id_token =
+        generate_id_token(audience.clone(), client_email.to_string(), service_account).await?;
+
+    let source_token_response_body = serde_json::json!({
+        "id_token": id_token,
+    })
+    .to_string();
+
+    let server = Server::run();
+    server.expect(
+        Expectation::matching(all_of![
+            request::method_path("GET", "/source_token"),
+            request::headers(contains(("metadata", "True",))),
+        ])
+        .respond_with(status_code(200).body(source_token_response_body)),
+    );
+
+    let contents = serde_json::json!({
+      "type": "external_account",
+      "audience": audience,
+      "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+      "token_url": "https://sts.googleapis.com/v1/token",
+      "credential_source": {
+        "url": server.url("/source_token").to_string(),
+        "headers": {
+          "Metadata": "True"
+        },
+        "format": {
+          "type": "json",
+          "subject_token_field_name": "id_token"
+        }
+      }
+    })
+    .to_string();
+
+    // Create external account with Url sourced creds
+    let creds =
+        ExternalAccountCredentialsBuilder::new(serde_json::from_str(contents.as_str()).unwrap())
+            .build()
+            .unwrap();
+
+    // Construct a BigQuery client using the credentials.
+    // Using BigQuery as it doesn't require a billing account.
+    let client = DatasetService::builder()
+        .with_credentials(creds)
+        .build()
+        .await?;
+
+    // Make a request using the external account credentials
+    client
+        .list_datasets()
+        .set_project_id(project)
+        .send()
+        .await?;
+
+    Ok(())
+}
+
+/// Generates a Google ID token using the iamcredentials generateIdToken API.
+/// https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials#sa-credentials-oidc
+async fn generate_id_token(
+    audience: String,
+    client_email: String,
+    service_account: serde_json::Value,
+) -> anyhow::Result<String> {
+    let creds = AccessTokenCredentialBuilder::new(service_account.clone())
+        .with_scopes(["https://www.googleapis.com/auth/cloud-platform"])
```

Comment:
Why set this? That is the default, right?

Reply:
Yes we default to cloud platform scope: 

https://github.com/googleapis/google-cloud-rust/blob/441aa4c20c08c6bb470f4ccb030623cf54d8a9cd/src/auth/src/credentials/service_account.rs#L222

Reply:
good call, I was trying to be explicit, but is definitely unnecessary. Just removed it